src/common/headers.rs

```diff
@@ -1,11 +1,15 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+use std::collections::HashMap;
 use std::result::Result;
 
 use crate::HttpHeaderError;
 use crate::RequestError;
 
+/// Maximum custom header lines in a request.
+const MAX_CUSTOM_HEADER_LINES: usize = 20;
```

Comment:
nit: Why 20? Is this simply a guesstimation? I would add some details to the comment about this number so it's not confused for some architecture limitation or anything.

Reply:
It's not an architecture limitation, but simply a raw estimate. I will make the comment more explicit.